fedora_messaging/cli.py

```diff
@@ -155,3 +164,120 @@ def consume(exchange, queue_name, routing_key, callback, app_name):
                 str(e.reason),
             )
             sys.exit(e.exit_code)
+
+
+@cli.command()
+@click.option("--publish-exchange", help=_publish_exchange_help)
+@click.option("--file", help=_publish_file_help, required=True, type=click.File("r"))
+def publish(publish_exchange, file):
+    """
+    Send messages from provided file to an AMQP queue.
+    """
+    msgs_json_str = file.read()
+
+    try:
+        messages = loads(msgs_json_str)
+    except exceptions.ValidationError as e:
+        _log.error("Unable to validate message: %s", str(e))
+        sys.exit(1)
```

Comment:
With each of these failure cases exiting as 1, it'll be difficult for scripts to handle specific error scenarios. If a file is invalid, no amount of retrying will help, but if a connection can't be established, waiting a bit might help. I recommend adding different exit codes for the different cases and documenting those codes.

Here I think you can just raise https://click.palletsprojects.com/en/7.x/api/#click.BadArgumentUsage or similar with the message you're logging and it'll do the right thing.